spec/lib/services/swagger-api-service-spec.js

```diff
@@ -502,6 +502,17 @@ describe('Services.Http.Swagger', function() {
             });
         });
 
+        it('should throw 500 on invalid template', function() {
+            res.body = { message: "foo" };
+            return renderer(req, res, 'foo', mockNext)
+            .then(function() {
+                expect(mockNext).to.be.calledWithMatch({status: 500});
+            },
+            function(err) {
```

Comment:
Hi @brianparry:

Below is a very simple test, I let the test case returns a rejected promise, it is surely caught by mocha, and other test cases are not impacted.

```javascript
it('return a rejected promise', function() {
    return Promise.reject();
});

it('return a fulfilled promise', function() {
    return Promise.resolve();
});
```

Below is the test output:
```
  test
    1) return a rejected promise
    ✓ return a fulfilled promise


  1 passing (7ms)
  1 failing

  1) test return a rejected promise:
     Error: Promise rejected with no or falsy reason
      at tryCatcher (node_modules/bluebird/js/main/util.js:26:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/main/promise.js:507:31)
      at Promise._settlePromiseAt (node_modules/bluebird/js/main/promise.js:581:18)
      at Promise._settlePromiseAtPostResolution (node_modules/bluebird/js/main/promise.js:245:10)
      at Async._drainQueue (node_modules/bluebird/js/main/async.js:128:12)
      at Async._drainQueues (node_modules/bluebird/js/main/async.js:133:10)
      at Async.drainQueues (node_modules/bluebird/js/main/async.js:15:14)
```